service/controller/resource/monitoring/prometheus/resource.go

```diff
@@ -246,9 +246,14 @@ func toPrometheus(ctx context.Context, v interface{}, config Config) (metav1.Obj
 			EvaluationInterval: promv1.Duration(config.EvaluationInterval),
 			Retention:          promv1.Duration(config.RetentionDuration),
 			RetentionSize:      promv1.ByteSize(config.RetentionSize),
+			// Fetches Prometheus rules from any namespace on the Management Cluster
+			// using https://v1-22.docs.kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-metadata-name
 			RuleNamespaceSelector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"name": key.NamespaceMonitoring(),
+				MatchExpressions: []metav1.LabelSelectorRequirement{
```

Comment:
I don't understand the point in just checking if this label exists.
As far as I understand, this label should exist on all namespaces, right?

Reply:
Yes the idea is that we check for rules in all namespaces as this label is set by Kubernetes, because otherwise prometheus operator will only check for rules in the namespace where prometheus is located and not the whole cluster